app/controllers/media/objects_controller.rb

```diff
@@ -47,7 +53,7 @@ def media_object_params
 
   def media_filename
     extension = File.extname(@media_object.item_file_name)
-    "elmo-#{@response.id}-#{@answer.id}#{extension}"
+    "elmo-#{@response.try(:id)}-#{@answer.try(:id)}#{extension}"
```

Comment:
The filename will be `elmo--.extension` which is weird but not broken and won't break anything -- I'm not sure how friendly we need to make the user experience of re-downloading a media file that you just uploaded to an invalid response, but if response and/or answer are nil, that method fails for the normal  `show` case for thumbnails which _does_ need to work when the response has not been saved yet.